cmd/minikube/cmd/docker-env.go

```diff
@@ -596,6 +607,20 @@ func tryDockerConnectivity(bin string, ec DockerEnvConfig) ([]byte, error) {
 	return c.CombinedOutput()
 }
 
+func dockerEnvSupported(containerRuntime, driverName string) bool {
+	if containerRuntime != constants.Docker && containerRuntime != constants.Containerd {
+		return false
+	}
+	if runtime.GOARCH != "amd64" && runtime.GOARCH != "arm64" {
+		return false
+	}
+	// we only support containerd-env on the Docker driver
+	if containerRuntime == constants.Containerd && driverName != driver.Docker {
+		return false
+	}
+	return true
```

Comment:
```suggestion
func dockerEnvSupported(containerRuntime, driverName string) error {
	if runtime.GOARCH != "amd64" && runtime.GOARCH != "arm64" {
		return fmt.Errorf("the docker-env command only supports amd64 & arm64 architectures.")
	}
	if containerRuntime != constants.Docker && containerRuntime != constants.Containerd {
		return fmt.Errorf("the docker-env command only supports the docker and containerd runtimes.")
	}
	// we only support containerd-env on the Docker driver
	if containerRuntime == constants.Containerd && driverName != driver.Docker {
		return fmt.Errorf("the docker-env command only supports the containerd runtime with the docker driver.")
	}
	return nil
```
To prevent the error message on line 294 from getting too long and confusing, we should change the message depending on what requirement that don't meet.